new-chatbot-app/src/database/database.service.ts

```diff
@@ -12,6 +12,30 @@ import { PrismaService } from './prisma-service/prisma.service';
 export class DatabaseService {
   private readonly logger = new Logger(DatabaseService.name);
   constructor(private readonly prismaService: PrismaService) {}
+
+  /**
+   * Get room ID from the database by fuzzy matching the roomCodeName and exact matching the buildingId
+   * @param roomCodeName
+   * @param buildingId
+   * @returns if no room satisfies the input condition, return null
+   */
+  async getRoomIdFromRoomCodeName(
+    roomCodeName: string,
```

Comment:
Could you explain why we use raw SQL query in this function rather than Prisma?[](url)

Reply:
Because Prisma doesn't have fuzzy matching itself, we have to rely on raw SQL query